api/post.go

```diff
@@ -405,6 +405,31 @@ func fireAndForgetNotifications(post *model.Post, teamId, teamUrl string) {
 					bodyPage.Props["PostMessage"] = model.ClearMentionTags(post.Message)
 					bodyPage.Props["TeamLink"] = teamUrl + "/channels/" + channel.Name
 
+					// attempt to fill in a message body based if the message has none
+					if len(strings.TrimSpace(bodyPage.Props["PostMessage"])) == 0 {
+						// extract the filenames from their paths and determine what type of files are attached
+						filenames := make([]string, len(post.Filenames))
+						onlyImages := true
+						for i, filename := range post.Filenames {
+							filenames[i] = strings.Replace(filepath.Base(filename), "+", " ", -1)
```

Comment:
It's just to make it look nicer in the resulting email. When testing, I noticed that "Image: Screen Shot 2015-07-10 at 10.25.30 AM.png sent" looked a lot nicer than "Image: Screen+Shot+2015-07-15+at+2.07.15+PM.png sent".